### PR TITLE
Implement RAC style library

### PR DESCRIPTION
**Pull request by helrond**
Implements the RAC style library.

A number of changes to the tests were also required. I'm not sure exactly why these were failing locally but not in CI but they are passing locally now.

I also removed some extraneous infrastructure that we don't use.

A couple of things to look at in the future:
- Updating the app to React 18, and updating the testing infrastructure at the same time.
- Looking at the overall structure of the app, and possibly bringing it more in line with how DIMES is structured.
- Reviewing the interaction pattern on the MapForm page - there are a lot of independent save actions (title, children, publish) that are happening which should probably just be combined into a single save method.

### CODE REVIEW

File: src/components/Modals.js [hunk 1]
```diff
@@ -59,108 +54,120 @@ export const MapComponentModal = ({ initialComponent, isOpen, onSubmit, path, to
...
+          X
+        </button>
+      </div>
+      <div className='modal-body p-20'>
+        { error
+          ? (
+            <div className="alert alert--orange" role="alert">
+            <div className="alert__text-wrapper">
+              <p className="alert__text">
+                {error}
```
Comment (HaSistrunk) [inline src/components/Modals.js]:
```suggestion
                Error message: {error}
```

**Comment by HaSistrunk** [comment]
I know this site is not widely used, but if we want to lock in some best practices, we should probably [stay away from disabled buttons](https://axesslab.com/disabled-buttons-suck/) as a rule. It's much better UX and more accessible to let someone click a button regardless, and then return a form error or whatever error message if they haven't entered the correct information instead of disabling the button until the user types something in. So that would apply to:
- "Save Title" button for adding a new map
- "Fetch from ArchivesSpace" when getting an arrangement map component in that modal
- "Save" when getting an arrangement map component in that modal

Probably something we should add to the Style Guide. That does require adding error handling here, though, so that's a choice and maybe something we want to wait and look at along with reviewing other interaction patterns.

File: src/components/MapList.js [hunk 2]
```diff
@@ -33,40 +35,43 @@ const MapList = () => {
...
+            <li key = {item.id}
+              className='mapList__item'>
+              <span className='mr-10'>{ item.title }</span>
+              <span>
+                <a href={`/maps/${item.id}`} className='btn btn--sm btn--blue mr-2'>Edit</a>
+                <Button
+                  onClick={() => toggleModal(item)}
+                  className='btn btn--sm btn--orange'
+                  label='Delete' />
+              </span>
```
Comment (HaSistrunk) [inline src/components/MapList.js]: 1. The "Edit" link and "Delete" button need to be programmatically associated with the arrangement map title that they act on. Adding an `aria-label` that gives them the name "Edit { item.title }" or "Delete { item.title }" would do the trick.

File: src/components/Modals.js [hunk 3]
```diff
@@ -59,108 +54,120 @@ export const MapComponentModal = ({ initialComponent, isOpen, onSubmit, path, to
...
+        <Button
+          className='btn--md btn--orange'
+          onClick={props.toggle}
+          label={props.cancelButtonText} />
+      </div>
+    </div>
   </Modal>
 )
 
 ConfirmModal.propTypes = {
```
Comment (HaSistrunk) [inline src/components/Modals.js:171]: The modal should close with the ESC key, so we'll want to add that option, which I see should be [available in reactstrap](https://reactstrap.github.io/?path=/docs/components-modal--modal).

File: src/components/Header.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        </div>
+        <nav className='nav-right' aria-label='Main'>
+          <ul className='nav__list'>
+            <li className="nav__item btn--navy">
+              <a className="nav__link" href="/maps/new">
+                Add New Map
+              </a>
+            </li>
+          </ul>
+        </nav>
```
Comment (HaSistrunk) [inline src/components/Header.js]: Is this actually navigation? I could see an argument that it is a main function of the site and should be moved out of the header nav and be, like, a large button after the `h1` similar to how the Edit Map page has the "Publish Map" button.

File: src/components/Modals.js (hunk 1 above)
Comment (HaSistrunk) [inline src/components/Modals.js]: I'm pretty certain that for `role=alert` to work, there has to be an empty div that already exists on the page with `role='alert'` (which is assertive and interrupts the screen reader) or `role='status'` (which is polite and waits for the screen reader to finish what it's doing and then reads the new content) assigned to it. Then, when the error appears dynamically, it is added to that div that is already in the DOM and the changes inside it are announced to the screen reader. This is something we need to update in the Style Guide (I [added an issue](https://github.com/RockefellerArchiveCenter/styles/issues/211)).

File: public/index.html [hunk 5]
```diff
@@ -24,6 +24,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
+    <link rel='stylesheet' href='https://assets.rockarch.org/v0.10.0/main.min.css'>
```
Comment (HaSistrunk) [inline public/index.html:27]: You'll also want to replace the local favicon with the one from the style library: https://assets.rockarch.org/assets/img/favicon.ico

**Comment by helrond** [comment]
Good call Hannah - to me that falls under reviewing interaction patterns because those disabled buttons are there in an attempt to make the current interaction patterns less breakable.

File: src/components/MapList.js [hunk 6]
```diff
@@ -33,40 +35,43 @@ const MapList = () => {
...
-            activeItem={activeMap}
-            toggle={toggleModal}
-            onConfirm={() => handleDelete(activeMap)}
-            message={`Are you sure you want to delete ${activeMap && activeMap.title}?`}
-            cancelButtonText = 'No, cancel'
-            confirmButtonText = 'Yes, delete' />
-        </div>
+    <>
+      <h1>Arrangement Maps</h1>
+      <div className='card'>
```
Comment (HaSistrunk) [inline src/components/MapList.js]:
```suggestion
      <div className='card card--container'>
```

File: src/index.css [hunk 7]
```diff
@@ -1,13 +1,24 @@
...
+.mapList {
+  width: 100%;
 }
+
+.mapList__item {
+  display: flex;
+  justify-content: space-between;
+  padding: 12px 20px;
+  border-bottom: 1px solid gray;
+}
```
Comment (HaSistrunk) [inline src/index.css]:
```suggestion
.mapList__item {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 12px 20px;
  border-bottom: 1px solid #5c5c5c;
}
```

**Review by HaSistrunk — CHANGES_REQUESTED** [review]
This is looking really good to me! I'm having trouble getting an arrangement map created because of fetching issues, so I haven't reviewed everything in the UI, but hopefully others can review that. A couple other things:
1. The modal size on desktop is pretty big for the content that's in there, particularly just for short confirmation messages. It may make sense to add styles to make those smaller to look a little better.
2. I'm not sure if this is an existing bug in Cartographer now, but I noticed if I go to "Add New Map", type in a title, and hit "Save", it displays the map on the homepage. If I'm understanding the logic correctly, we'd only want the new map to appear in the arrangement maps on the homepage once the map is published, right?

File: src/components/Modals.js (hunk 1 above)
Comment (helrond) [reply src/components/Modals.js]: @HaSistrunk I think I got this right, but wouldn't mind a double-check.

**Comment by helrond** [comment]
Thanks for your detailed review @HaSistrunk!

Regarding your second point above - yes, that is intended behavior. Maps should be able to be created independently of being published. If we didn't show the map until it was published there would be no way of editing a map once it was created if it wasn't immediately published.

File: src/components/Modals.js (hunk 1 above)
Comment (HaSistrunk) [reply src/components/Modals.js]: Yes! I tested and this works well.

**Review by HaSistrunk — APPROVED** [review]
Nice! Looks great to me.

**Review by p-galligan — APPROVED** [review]

**Review by kemartin2015 — APPROVED** [review]

**Review by daremiyo — APPROVED** [review]

**Pull request merged**